vio/viosslfactories.c

```diff
@@ -238,8 +386,14 @@ new_VioSSLFd(const char *key_file, const char *cert_file,
     NOTE: SSL_CTX_set_cipher_list will return 0 if
     none of the provided ciphers could be selected
   */
-  if (cipher &&
-      SSL_CTX_set_cipher_list(ssl_fd->ssl_context, cipher) == 0)
+  strcat(cipher_list, tls_cipher_blocked);
+  if (cipher)
+    strcat(cipher_list, cipher);
```

Comment:
Good catch :)

=================================================================
==19544==ERROR: AddressSanitizer: stack-buffer-overflow on address 0x7fff595e79e0 at pc 0x000108229555 bp 0x7fff595e69b0 sp 0x7fff595e6170
WRITE of size 4983 at 0x7fff595e79e0 thread T0
    #0 0x108229554 in wrap_strcat (libclang_rt.asan_osx_dynamic.dylib+0x41554)
    #1 0x107107b04 in new_VioSSLFd viosslfactories.c:391
    #2 0x1071081c4 in new_VioSSLAcceptorFd viosslfactories.c:504
    #3 0x10683a9c3 in init_ssl() mysqld.cc:3815
    #4 0x106836c8a in mysqld_main(int, char**) mysqld.cc:4775
    #5 0x7fff8e0d35ac in start (libdyld.dylib+0x35ac)
    #6 0x8  (<unknown module>)

Address 0x7fff595e79e0 is located in stack of thread T0 at offset 4128 in frame
    #0 0x10710783f in new_VioSSLFd viosslfactories.c:328

  This frame has 2 object(s):
    [32, 4128) 'cipher_list'
    [4256, 4288) '_db_stack_frame_' <== Memory access at offset 4128 partially underflows this variable